src/fmu/dataio/providers/_filedata.py

```diff
@@ -104,9 +106,9 @@ def _get_share_folders(self) -> Path:
         logger.info("Export share folders are %s", sharefolder)
         return sharefolder
 
-    def _add_filename_to_path(self, path: Path) -> Path:
+    def _get_filename(self) -> Path:
         stem = self._get_filestem()
-        return (path / stem).with_suffix(path.suffix + self.objdata.extension)
+        return Path(stem).with_suffix(self.objdata.extension)
```

Comment:
Slight refactoring and simplification of this method. The path provided here was only folders and never had a suffix.